acceptance/tests/windows/PUP-9719_windows_system_first_pa_run.rb

```diff
@@ -0,0 +1,53 @@
+test_name "PUP-9719 Windows First Agent run as SYSTEM sets cache file permissions correctly" do
+  tag 'risk:medium',
+    'audit:medium',
+    'audit:integration'
+
+  confine :to, :platform => 'windows'
+
+  require 'puppet/acceptance/temp_file_utils'
+  extend Puppet::Acceptance::TempFileUtils
+
+  agents.each do |agent|
+    statedir = on(agent, puppet('config print statedir')).stdout.chomp
+    client_datadir = on(agent, puppet('config print client_datadir')).stdout.chomp
+
+    teardown do
+      on agent, "schtasks /delete /tn PuppetSystemRun /F"
+      on agent, "rm -rf #{statedir}/*"
+      on agent, "rm -rf #{client_datadir}/catalog/*"
+    end
+
+    step "Clean the ProgramData cache directory first" do
+      on agent, "rm -rf #{statedir}/*"
+      on agent, "rm -rf #{client_datadir}/catalog/*"
+    end
+
+    step "Create and run a scheduled task on System Account." do
+      on agent, "schtasks /create /tn PuppetSystemRun /RL HIGHEST /RU SYSTEM /F /SC ONCE /ST 23:59 /TR 'cmd /c \"%ProgramFiles%\\Puppet Labs\\Puppet\\bin\\puppet.bat\" agent -t >> c:\\Windows\\Temp\\Puppet-System-Run.log 2>&1'"
```

Comment:
How does setting start time to "23:59" ensure the agent runs "now"? Based on https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/schtasks#BKMK_now, it sounds like we'd have to get the current time, and add a few seconds, but I think the time needs to be in the future, as if we "miss" the time, then it won't run until tomorrow?

Can we use `psexec` to run the agent now, instead of dealing with scheduled tasks?

Reply:
This just create the task, running the task now is done on the next line:
```on agent, "schtasks /run /tn PuppetSystemRun"```

Reply:
Are there timing issues where the task scheduler might run the task (if we're close to 23:59) and then we try to run the task again?